optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/view/IndictmentView.java

```diff
@@ -0,0 +1,87 @@
+package org.optaplanner.openshift.employeerostering.shared.roster.view;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+public class IndictmentView {
+
+    private List<Rules> ruleMatchedList;
+    private List<HardSoftScore> scoreList;
+
+    public IndictmentView() {
+
+    }
+
+    public IndictmentView(Indictment indictment) {
+        ruleMatchedList = new ArrayList<>(indictment.getConstraintMatchCount());
+        scoreList = new ArrayList<>(indictment.getConstraintMatchCount());
+
+        for (ConstraintMatch cm : indictment.getConstraintMatchSet()) {
+            ruleMatchedList.add(Rules.forConstraintMatch(cm));
+            scoreList.add((HardSoftScore) cm.getScore());
+        }
+    }
+
+    public IndictmentView(List<Rules> ruleMatched, List<HardSoftScore> score) {
+        this.ruleMatchedList = ruleMatched;
+        this.scoreList = score;
+    }
+
+    public List<Rules> getRuleMatchedList() {
+        return ruleMatchedList;
+    }
+
+    @JsonSerialize(contentUsing = ScoreJacksonJsonSerializer.class)
+    @JsonDeserialize(contentUsing = HardSoftScoreJacksonJsonDeserializer.class)
```

Comment:
This can now be removed

Reply:
Not yet, requires https://github.com/kiegroup/optashift-employee-rostering/pull/101